core/src/main/scala/kafka/coordinator/group/DelayedSync.scala

```diff
@@ -0,0 +1,49 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.coordinator.group
+
+import kafka.server.DelayedOperation
+
+/**
+ * Delayed rebalance operation that is added to the purgatory when is completing the rebalance.
+ *
+ * Whenever a SyncGroup is received, checks that we received all the SyncGroup request from
+ * each member of the group; if yes, complete this operation.
+ *
+ * When the operation has expired, any known members that have not sent a SyncGroup requests
+ * are removed from the group. If any members is removed, the group is rebalanced.
+ */
+private[group] class DelayedSync(
```

Comment:
There are a set of metrics that get exposed automatically for each purgatory we add. I'm considering if these will be useful or if we should add a flag to let us disable them.

Reply:
Ah.. I was not aware of this. This means that I cannot change the purgatory name like I did.

I think that it is useful to have some metrics. The issue is that we would need a KIP, right?

As an alternative, we could perhaps share the purgatory for the delayed join and the delayed sync. We could use different keys for instance. What do you think?

Reply:
@hachikuji I went ahead with the consolidating both purgatories into one. This way we can keep the same metrics as today and we avoid introducing a new purgatory.